test/controllers/hosts_controller_test.rb

```diff
@@ -1301,6 +1301,29 @@ def test_submit_multiple_rebuild_config_pessimistic
     assert_match(/host must have/, flash[:error])
   end
 
+  describe '#ipmi_boot' do
+    setup do
+      @request.env['HTTP_REFERER'] = host_path(@host.id)
+      setup_user 'ipmi_boot', 'hosts'
+    end
+
+    test 'responds gracefully if BMC is not available' do
+      put :ipmi_boot, { :id => @host.id, :ipmi_device => 'bios'},
+        set_session_user.merge(:user => @one.id)
+      assert_match(/No BMC NIC available for host/, flash[:error])
+      assert_redirected_to host_path(@host.id)
+    end
+
+    test 'responds gracefully if BMC is not available' do
```

Comment:
This test name should be changed, perhaps to say it responds correctly for a non-admin user or similar.